src/examples/algorithms/Anomaly.cpp

```diff
@@ -0,0 +1,199 @@
+        std::vector<std::string> tokens(tok.begin(), tok.end());
+
+        // Skip line if it the aren't 2 elements at least.
+        if (std::distance(tok.begin(), tok.end()) < 1) continue;
+
+        // Store both items
+        column1.push_back(std::stof(tokens[0]));
+    }
+
+    std::cout << "Column1: ";
+    print_vector(column1, ' ');
+    std::cout << std::endl;
+
+    ScalarEncoder encoder{
+        25,   // w: Active bits per value
+        0,   // min value
+        10,  // max value
+        250,  // n: bits per output value,
+        0,
+        0,
+        true // Clip input to min/max vlues if needed.
+    };
+
+    // Spatial Pooler
+    SpatialPooler sp{
+        std::vector<UInt>{encoder.getOutputWidth()}, // Input dimensions = encoder width
+        std::vector<UInt>{COLS}  // Number of colums per dimension
+    };
+
+    // Temporal Pooler
+    Cells4 tp{
+        COLS,    // Number of columns
+        CELLS,     // Cells per column
+        12, 
+        8, 
+        15, 
+        5, 
+        .5, 
+        .8, 
+        1.0, 
+        .1, 
+        .1, 
+        0.0, 
+        false, 
+        42, 
+        true, 
+        false
+
+    };
+
+    std::vector<UInt> spOutput(COLS);
+    std::vector<UInt> lastTPOutput;
+
+    int iteration = 0;
+    for (auto& elem: column1) {
+        auto scalar = std::vector<Real32>(encoder.getOutputWidth());
+        encoder.encodeIntoArray(elem, scalar.data());
+
+        // For some reason, the ScalarEncoder outputs a Real32 vector,
+        // which is incomaptible with the SpatialPooler's input, hence a new 
+        // vector is created with the required type, and the elements
+        // are converted.
+        auto uint_scalar = vectorTypeCast<Real32, UInt>(scalar);
+        std::cout << elem << ":\n";
+        std::cout << "Scalar encoder: ";
+        print_vector(uint_scalar);
+
+        sp.compute(uint_scalar.data(), true, spOutput.data());
+        std::cout << "Output of spatial pooler: ";
+        print_vector(spOutput);
+        std::cout << std::endl;
+
+
+        // The temporal pooler uses Real32, so the vector must be converted again
+        auto tpInput = vectorTypeCast<UInt, Real32>(spOutput);
+        std::vector<Real32> tpOutput(COLS*CELLS);
+        tp.compute(tpInput.data(), tpOutput.data(), true, true);
+
+        // And the result is converted ONCE again to UInts for pretty printing to stdout.
+        auto uintTpOutput = vectorTypeCast<Real32, UInt>(tpOutput);
+        std::cout << "Output of temporal pooler: ";
+        print_vector(uintTpOutput);
+        std::cout << std::endl;
+
+        if (iteration > 0) {
+            // TODO: Anomaly...
+            Anomaly a(0, AnomalyMode::PURE, 0);
+            float c = a.compute(
+                binaryToSparse(spOutput),     // Spatial pooler (current active columns)
+                binaryToSparse(lastTPOutput), // Temporal pooler (previously predicted columns)
```

Comment:
what are the correct vectors to feed to Anomaly here? 
I'm aware this is not correct, it should be something like TM.getActiveColumns() and TM.getPredictedColumns() at time T-1; 

TM has only getActive/PredicteveCells().. (see the issue for more details please)